### DISCUSSION TITLE
Current locale inside getInitialProps

### DISCUSSION THREAD

**Discussion by botond-veress** [vercel/next.js#20242] [Help]
Hey everyone. 👋 

Is it possible to get the current locale in `getInitialProps`? In the docs it is mentioned that the locale is added to the `context` for `getStaticProps` and `getServerSideProps`. Unfortunately, I have to stick with `getInitialProps` for now because `getServerSideProps` doesn't play well with GraphQL (I just couldn't figure it out how to persist the GraphQL cache during SSR).
If it's not possible yet, are there any plans to add the current locale it to the `getInitialProps`'s `context` in the future?

Thank you

**Comment by botond-veress** [comment]
Actually, I was able to forward the `locale` to the `getInitialsProps` by adding a `getInitialsProps` to the app as well.
```typescript
export default class App extends NextApp<Props, {}, ComponentProps> {
  // ...
  static getInitialProps = async ({ Component, router, ctx }: AppContext) => {
    const locale = normalizeLanguage(router.locale);

    const pageContext = { ...ctx, locale };
    const pageProps = Component.getInitialProps ? await Component.getInitialProps(pageContext) : {};

    return { pageProps };
  };
  // ...
}
```

It would be great if this would be done automatically as it is done for SSR/SSG.

**Reply by samyakbhaleraogv** [reply]
@botond-veress : So we are calling getInitialProps for that component on server side and injecting locale. This results in getInitialProps to have locale when it gets executed on client side. Is my understanding correct ?
